cluster/userdata-master.yaml

```diff
@@ -853,6 +853,59 @@ write_files:
               securityContext:
                 privileged: true
 
+
+  - path: /srv/kubernetes/manifests/daemonsets/cluster-autoscaler.yaml
+    content: |
+      apiVersion: extensions/v1beta1
+      kind: Deployment
+      metadata:
+        name: cluster-autoscaler
+        namespace: kube-system
+        labels:
+          app: cluster-autoscaler
+      spec:
+        replicas: 1
+        selector:
+          matchLabels:
+            app: cluster-autoscaler
+        template:
+          metadata:
+            labels:
+              app: cluster-autoscaler
+            annotations:
+              scheduler.alpha.kubernetes.io/critical-pod: ''
+          spec:
+            hostNetwork: true
```

Comment:
it looks like there are problem with this approach. The SDK has an hard time determining the role, even though everything seems good.

````
E1104 14:41:04.846531       1 aws_cloud_provider.go:264] Error describing EC2 instances: NoCredentialProviders: no valid providers in chain
F1104 14:41:04.846555       1 cluster_autoscaler.go:167] Failed to create AWS cloud provider: NoCredentialProviders: no valid providers in chain
````